lib/client/api.go

```diff
@@ -1088,6 +1088,32 @@ func Status(profileDir, proxyHost string) (*ProfileStatus, []*ProfileStatus, err
 	return profileStatus, others, nil
 }
 
+// FetchRoleSet fetches a user's roles for a specified cluster.
+func FetchRoleSet(ctx context.Context, log *logrus.Entry, cluster auth.ClientI, profile *ProfileStatus) (services.RoleSet, error) {
+	// get roles and traits. default to the set from profile, try to get up-to-date version from server point of view.
+	roles := profile.Roles
+	traits := profile.Traits
+
+	// GetCurrentUser() may not be implemented, fail gracefully.
+	user, err := cluster.GetCurrentUser(ctx)
+	if err == nil {
+		roles = user.GetRoles()
+		traits = user.GetTraits()
+	} else {
+		log.Debugf("Failed to fetch current user information: %v.", err)
```

Comment:
Is passing the logger necessary? Seems strange to special case this one function.